monai/losses/aucm_loss.py

```diff
@@ -0,0 +1,172 @@
+# Copyright (c) MONAI Consortium
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#     http://www.apache.org/licenses/LICENSE-2.0
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import warnings
+from typing import Callable, List, Optional, Union
+
+import torch
+from torch.nn.modules.loss import _Loss
+
+from monai.networks import one_hot
+from monai.utils import LossReduction
+
+
+class AUCMLoss(_Loss):
+    """
+    AUCM Loss with squared-hinge function: a novel loss function to directly optimize AUROC
+
+    The original paper: Large-scale Robust Deep AUC Maximization: A
+    New Surrogate Loss and Empirical Studies on Medical Image Classification
+    https://arxiv.org/abs/2012.03173
+    """
+
+    def __init__(
+        self,
+        margin: float = 1.0,
+        imratio: List[float] = None,
+        num_classes: int = 2,
+        to_onehot_y: bool = False,
+        sigmoid: bool = False,
+        softmax: bool = False,
+        other_act: Optional[Callable] = None,
+        reduction: Union[LossReduction, str] = LossReduction.MEAN,
+        gpu: bool = False,
```

Comment:
Maybe remove `gpu` args will be good. The device is typically determined by the input.